include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -105,199 +149,206 @@ class CSummedAreaTableImageFilter : public CMatchedSizeInOutImageFilterCommon, p
-								const auto outDataAdressOffsetScratch = globalIndependentOffset * currentChannelCount;
+					auto normalizeScratch = [&]()
+					{
+						core::vector3du32_SIMD localCoord;
+							for (auto& z = localCoord[2] = 0u; z < extent.z; ++z)
+								for (auto& y = localCoord[1] = 0u; y < extent.y; ++y)
+									for (auto& x = localCoord[0] = 0u; x < extent.x; ++x)
+									{
+										const auto scratchInPlaneOffset = ((localCoord.z * state->extent.height + localCoord.y) * state->extent.width + localCoord.x);
+										const auto outDataAdressOffsetScratch = scratchInPlaneOffset * currentChannelCount;
 
-								auto* entryScratchAdress = scratchMemory + globalTranslatedDecodeTypeRegionOffset + outDataAdressOffsetScratch;
-								
-								for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
-									*(entryScratchAdress + channel) /= *(reinterpret_cast<decodeType*>(imageTotalCacheBuffers[localCoord.z]->getPointer()) + channel);     
-							}
-				};
+										auto* entryScratchAdress = scratchMemory + outDataAdressOffsetScratch;
+										auto* entryTotalImageCacheValues = reinterpret_cast<decodeType*>(totalImageCacheScratch + (localCoord.w * state->extent.depth + localCoord.z) * state->mainRowScratchCacheByteSize);
 
-				auto encodeTo = [&](ICPUImage* outImage)
-				{
-					core::vector3du32_SIMD localCoord;
-					for (auto& z = localCoord[2] = 0u; z < trueExtent.z; ++z)
-						for (auto& y = localCoord[1] = 0u; y < trueExtent.y; ++y)
-							for (auto& x = localCoord[0] = 0u; x < trueExtent.x; ++x)
-							{
-								const auto globalIndependentOffset = ((localCoord.z * trueExtent.y + localCoord.y) * trueExtent.x + localCoord.x);
-								const auto outDataAdressOffsetInput = globalIndependentOffset * texelByteSize;
-								const auto outDataAdressOffsetScratch = globalIndependentOffset * currentChannelCount;
+										for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
+											entryScratchAdress[channel] /= entryTotalImageCacheValues[channel];
+									}
+					};
+					*/
 
-								auto* entryScratchAdress = scratchMemory + globalTranslatedDecodeTypeRegionOffset + outDataAdressOffsetScratch;
-								auto* outData = commonExecuteData.outData + outDataAdressOffsetInput; // TODO need to use filters API to stay at layer-mipmap memory
-								asset::encodePixelsRuntime(commonExecuteData.inFormat, outData, entryScratchAdress);
-							}
-				};
+					/*
 
-				if(state->normalizeImageByTotalSATValues)
-					if (!asset::isNormalizedFormat(commonExecuteData.inFormat))
-						normalizeScratch();
+					it wont work anymore since now
 
-				encodeTo(commonExecuteData.outImg);
-				return true;
-			};
+					if (state->normalizeImageByTotalSATValues)
+						if (!asset::isNormalizedFormat(inFormat))
+							normalizeScratch();
 
-			return commonExecute(state, perOutputRegion);
+					*/
+
+					return commonExecute(state, perOutputRegionEncode);
+				}
+
+				_IRR_ALIGNED_FREE(stateCopy.scratchMemory);
```

Comment:
WHY WHY WHY ARE YOU STILL ALLOCATING SCRATCH MEMORY?

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/497#discussion_r451802608

```C++
auto stateCopy = state;
```

Reply:
I don't get the pseudo code https://github.com/buildaworldnet/IrrlichtBAW/pull/497#discussion_r451822733 and I think that https://github.com/buildaworldnet/IrrlichtBAW/pull/497#discussion_r463598904
I don't get it how I can touch `w` otherwise, there must be 2 scratches at ones holding single layer then probably

Reply:
corrected in following commit